cns/middlewares/swiftV2.go

```diff
@@ -0,0 +1,169 @@
+package middlewares
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/types"
+	"github.com/Azure/azure-container-networking/crd/multitenancy/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	errMTPNCNotReady         = errors.New("mtpnc is not ready")
+	errInvalidSWIFTv2NICType = errors.New("invalid NIC type for SWIFT v2 scenario")
+)
+
+const (
+	prefixLength     = 32
+	overlayGatewayv4 = "169.254.1.1"
+	overlayGatewayV6 = "fe80::1234:5678:9abc"
+)
+
+type SWIFTv2Middleware struct {
+	Cli client.Client
+}
+
+// ValidateIPConfigsRequest validates if pod is multitenant by checking the pod labels, used in SWIFT V2 scenario.
+// nolint
+func (m *SWIFTv2Middleware) ValidateIPConfigsRequest(ctx context.Context, req *cns.IPConfigsRequest) (respCode types.ResponseCode, message string) {
+	// Retrieve the pod from the cluster
+	podInfo, err := cns.UnmarshalPodInfo(req.OrchestratorContext)
+	if err != nil {
+		errBuf := fmt.Sprintf("unmarshalling pod info from ipconfigs request %v failed with error %v", req, err)
+		return types.UnexpectedError, errBuf
+	}
+	podNamespacedName := k8stypes.NamespacedName{Namespace: podInfo.Namespace(), Name: podInfo.Name()}
+	pod := v1.Pod{}
+	if err := m.Cli.Get(ctx, podNamespacedName, &pod); err != nil {
+		errBuf := fmt.Sprintf("failed to get pod %v with error %v", podNamespacedName, err)
+		return types.UnexpectedError, errBuf
+	}
+
+	// check the pod labels for Swift V2, set the request's SecondaryInterfaceSet flag to true.
+	if _, ok := pod.Labels[configuration.LabelPodSwiftV2]; ok {
+		req.SecondaryInterfacesExist = true
+	}
+	return types.Success, ""
+}
+
+// GetIPConfig returns the pod's SWIFT V2 IP configuration.
+func (m *SWIFTv2Middleware) GetIPConfig(ctx context.Context, podInfo cns.PodInfo) (cns.PodIpInfo, error) {
+	// Check if the MTPNC CRD exists for the pod, if not, return error
+	mtpnc := v1alpha1.MultitenantPodNetworkConfig{}
+	mtpncNamespacedName := k8stypes.NamespacedName{Namespace: podInfo.Namespace(), Name: podInfo.Name()}
+	if err := m.Cli.Get(ctx, mtpncNamespacedName, &mtpnc); err != nil {
+		return cns.PodIpInfo{}, fmt.Errorf("failed to get pod's mtpnc from cache : %w", err)
```

Comment:
I noticed that the package `github.com/pkg/errors` have been made obsolete since go 1.13 (see answer [here](https://stackoverflow.com/questions/61933650/whats-the-difference-between-errors-wrapf-errors-errorf-and-fmt-errorf), and also the repo has been archived since [2021](https://github.com/pkg/errors). Should we still be using it?

Reply:
we are still using it.